common/src/main/java/com/wynntils/models/lootrun/LootrunModel.java

```diff
@@ -158,6 +167,12 @@ public class LootrunModel extends Model {
 
     private Map<BeaconColor, Integer> selectedBeacons = new TreeMap<>();
 
+    @Persisted
+    private final Storage<Map<String, List<MissionType>>> missionStorage = new Storage<>(new TreeMap<>());
+
+    private final Pattern[] missionPatterns = new Pattern[2];
+    private final StyledText[] recentChatMessages = new StyledText[5];
```

Comment:
I am not happy about either of these, but especially about `recentChatMessages`. Let's keep this comment here, and we can get back to it, when most of the other requests are done, and the code is finalising.

Reply:
If it wasn't clear from the code, the recent chat messages are required to identify that a mission has been completed instead of starting. Please let me know if you can think of a better way to achieve this. I do agree that the missionPatterns array is a little sketchy and could probably be done in a better way.

Reply:
You can do exactly what I did above your code, to get rid of `recentChatMessages`.  If `LOOTRUN_COMPLETED_PATTERN/LOOTRUN_FAILED_PATTERN` matches, then a flag is set, that expect the next message to contain fail/complete data. You can do the same thing here, set an "expectMission"/"expectMissionComplete" flag, and the next time you parse a mission from chat, you know what it is.

Secondly, the added code around `onChatMessage` needs changes. At first, I had no idea why it even worked. I see now. `StyledText#toString` does not return a string representation of the text. If you check the method, it's clear that it's a debug-style text. You have multiple `getString` methods available for that. Then, you can replace all the `matcher.find()` calls with `"matcher.matches()`.

Reply:
I have taken your advice and replaced the recentChatMessages system with one with a flag that expects mission completion. Would you still like me to change the missionPatterns array? It's currently in an array to avoid adding too many variables at the top of the file, but since I don't anticipate any more entries being added in the future, I could change it to a simple two-variable solution.